v4/core/base_service.go

```diff
@@ -451,3 +480,157 @@ func getErrorMessage(responseMap map[string]interface{}, statusCode int) string
 	// for the status code.
 	return http.StatusText(statusCode)
 }
+
+// EnableRetries will construct a "retryable" HTTP Client with the specified
+// configuration, and then set it on the service instance.
+// If maxRetries and/or maxRetryInterval are specified as 0, then default values
+// are used instead.
+func (service *BaseService) EnableRetries(maxRetries int, maxRetryInterval time.Duration) {
+	client := NewRetryableHTTPClient()
+	if maxRetries > 0 {
+		client.RetryMax = maxRetries
+	}
+	if maxRetryInterval > 0 {
+		client.RetryWaitMax = maxRetryInterval
+	}
+
+	service.SetHTTPClient(client.StandardClient())
```

Comment:
The `StandardClient()` method returns an http.Client instance with the retryable client "hidden" in the transport.   This means that the BaseService.Client field will always hold an http.Client instance.   If it's a retryable client, then the transport holds the actual go-retryablehttp.Client instance.